sdk/go/common/workspace/plugins.go

```diff
@@ -1005,99 +1006,151 @@ type PluginSpec struct {
 
 type PluginVersionNotFoundError error
 
+var urlRegex = sync.OnceValue(func() *regexp.Regexp {
+	return regexp.MustCompile(`^[^\./].*\.[a-z]+/[a-zA-Z0-9-/]*[a-zA-Z0-9/](@.*)?$`)
+})
+
+var gitCommitRegex = sync.OnceValue(func() *regexp.Regexp {
```

Comment:
Since we're making this global, let's maybe call it `gitCommitHashRegex`?  Also the commment that this supports sha1 and sha256 hashes should probably be preserved.